src/account/operations/transfer/high_level/mint_native_token.rs

```diff
@@ -152,9 +155,9 @@ impl AccountHandle {
             )?])?;
 
             let outputs = vec![
-                Output::Alias(new_alias_output_builder.finish()?),
-                Output::Foundry(
-                    FoundryOutputBuilder::new_with_amount(
+                new_alias_output_builder.finish_output()?,
+                {
+                    let mut foundry_builder = FoundryOutputBuilder::new_with_amount(
                         minimum_storage_deposit_foundry(&byte_cost_config)?,
```

Comment:
I think we need to change this, if there is `foundry_metadata`
And what I'm also not sure is if we should increase the storage deposit so the foundry can mint native tokens without further inputs (add the required storage deposit amount for one NativeTokens)

Reply:
Regarding the first sentence you are right. But in https://github.com/iotaledger/wallet.rs/pull/1007 I'm removing `minimum_storage_deposit_foundry` and using `new_with_minimum_storage_deposit` so it should take care of the metadata to compute the appropriate amount. I'm not sure what you mean with the second sentence though, can you elaborate?

Reply:
For the minted native tokens we currently add a new basic output which will get them, but if the foundry would have a larger amount, we could also add the native tokens to the foundry output directly, that way we have one output less (and if we don't mint any native tokens in this call, we could the next time use only the alias and foundry output as input and mint the native tokens to the foundry output)

Reply:
Just a possible optimization, not really required

Reply:
Well at least the first sentence is an actual mistake

Reply:
Yes, but if it's fixed in https://github.com/iotaledger/wallet.rs/pull/1007 it should be fine